### PR TITLE
[docs] [migrate] Add a troubleshoot page

### PR DESCRIPTION
**Pull request by aishwarya24**
Summary:
- resolves https://github.com/yugabyte/yb-voyager/issues/439
- Ordered the issues by MySQL, Oracle,  and then both.
- Moved unsupported features from yb-voyager-cli page to the troubleshoot page.

Build preview : https://deploy-preview-15097--infallible-bardeen-164bc9.netlify.app/preview/migrate/troubleshoot/

### CODE REVIEW

File: docs/content/preview/migrate/troubleshoot.md [hunk 1]
```diff
@@ -0,0 +1,426 @@
...
+    float_type double precision,
+    double_type double precision,
+    double_precision_type double precision,
+    real_type double precision
+);
+```
+
+Suggested changes to the schema can be done using one of the following options:
+
+1. Edit the `/etc/yb-voyager/base-ora2pg.conf` file before exporting schema, using the `DATA_TYPE` directive.
```
Comment (ddhodge) [inline docs/content/preview/migrate/troubleshoot.md:97]: use bullets; numbers seem too much like steps

File: docs/content/preview/migrate/troubleshoot.md [hunk 2]
```diff
@@ -0,0 +1,426 @@
...
+type: docs
+---
+
+This page documents the known issues and workarounds, and unsupported features when migrating data with YugabyteDB Voyager.
+
+## Known issues
+
+### Issue #573: [MySQL] Unsigned decimal types are not exported properly
+
+**GitHub link**: [Issue #573](https://github.com/yugabyte/yb-voyager/issues/573)
```
Comment (ddhodge) [inline docs/content/preview/migrate/troubleshoot.md:20]: some issues are closed. what does that mean

File: docs/content/preview/migrate/troubleshoot.md [hunk 3]
```diff
@@ -0,0 +1,426 @@
...
+
+---
+
+### Issue #579: [MySQL] Functional/Expression indexes fail to migrate
+
+**GitHub link**: [Issue #579](https://github.com/yugabyte/yb-voyager/issues/579)
+
+**Description**: If your schema contains Functional/Expression indexes in MYSQL, the index creation fails with a syntax error during migration and doesn't get migrated.
+
+**Workaround**: Manual intervention needed. You have to remove the back-ticks (``) or oblique quotes ("") to the exported schema files.
```
Comment (ddhodge) [inline docs/content/preview/migrate/troubleshoot.md:140]: from the exported

File: docs/content/preview/migrate/troubleshoot.md [hunk 4]
```diff
@@ -0,0 +1,426 @@
...
+CREATE INDEX exp_ind ON exp_index_test ((extract(year from date(to_date))));
+```
+
+---
+
+### Issue #320: [MySQL] Issue when exporting data from MySQL with table_name including quotes
+
+**GitHub link**: [Issue #320](https://github.com/yugabyte/yb-voyager/issues/320)
+
+**Description**: When exporting the schema from MySQL which includes the table_name with quotes, it exports the table with the table_name converted to lowercase and without quotes, resulting in an error when exporting the data.
```
Comment (ddhodge) [inline docs/content/preview/migrate/troubleshoot.md:168]:
```suggestion
**Description**: When exporting a schema from MySQL that includes a table name that has quotes, the table is exported with the table name converted to lowercase and without the quotes, resulting in an error.
```

File: docs/content/preview/migrate/troubleshoot.md [hunk 5]
```diff
@@ -0,0 +1,426 @@
...
+
+**Workaround**: Manual intervention needed. You have to rename the table in MySQL and then export and import the data followed by renaming it in YugabyteDB.
```
Comment (ddhodge) [inline docs/content/preview/migrate/troubleshoot.md:170]:
```suggestion
**Workaround**: Manual intervention needed. You have to rename the table in MySQL and then export and import the data followed by renaming the table in YugabyteDB.
```

File: docs/content/preview/migrate/troubleshoot.md [hunk 6]
```diff
@@ -0,0 +1,426 @@
...
+```output
++----------------------+
+| Tables_in_pk_missing |
++----------------------+
+| "test_data_COPY"     |
+| TEST_DATA            |
++----------------------+
+```
+
+The exported schema eaxmple is as follows:
```
Comment (ddhodge) [inline docs/content/preview/migrate/troubleshoot.md:187]:
```suggestion
The exported schema is as follows:
```

File: docs/content/preview/migrate/troubleshoot.md [hunk 7]
```diff
@@ -0,0 +1,426 @@
...
+Error when exporting data is as follows:
+
+```output
+DBD::mysql::st execute failed: Table 'pk_missing.test_data_COPY' doesn't exist at /usr/local/share/perl5/Ora2Pg.pm line 14247.
+DBD::mysql::st execute failed: Table 'pk_missing.test_data_COPY' doesn't exist at /usr/local/share/perl5/Ora2Pg.pm line 14247.
+```
+
+Suggested workaround is as follows:
+
+1. Rename the table name with quotes to a name without quotes in MySQL database using the following command:
```
Comment (ddhodge) [inline docs/content/preview/migrate/troubleshoot.md:210]:
```suggestion
1. Rename the table with the quotes using a name without quotes in MySQL database using the following command:
```

File: docs/content/preview/migrate/troubleshoot.md [hunk 8]
```diff
@@ -0,0 +1,426 @@
...
+
+    ```sql
+    Alter table `"test_data_COPY"` rename test_data_COPY2;
+    ```
+
+1. Export and import the data.
+
+1. Rename the table name in YugabyteDB to a quoted one using the following command:
```
Comment (ddhodge) [inline docs/content/preview/migrate/troubleshoot.md:218]:
```suggestion
1. Rename the table in YugabyteDB to include the quotes using the following command:
```

File: docs/content/preview/migrate/troubleshoot.md [hunk 9]
```diff
@@ -0,0 +1,426 @@
...
+    Alter table test_data_copy2 rename to "test_data_COPY";
+    ```
+
+---
+
+### Issue #207: [Oracle] Some numeric types do not get exported
+
+**GitHub link**: [Issue #207](https://github.com/yugabyte/yb-voyager/issues/207)
+
+**Description**: For cases where the precision is less than the scale in a numeric attribute, it fails to get imported to YugabyteDB.
```
Comment (ddhodge) [inline docs/content/preview/migrate/troubleshoot.md:230]: what is 'it'?

File: docs/content/preview/migrate/troubleshoot.md [hunk 10]
```diff
@@ -0,0 +1,426 @@
...
+
+**Workaround**: Manually remove the explicit precision and scale values from the exported numeric or decimal attributes. PostgreSQL or YugabyteDB does not allow setting the precision less than the scale explicitly.
```
Comment (ddhodge) [inline docs/content/preview/migrate/troubleshoot.md:232]:
```suggestion
**Workaround**: Manually remove the explicit precision and scale values from the exported numeric or decimal attributes. PostgreSQL and YugabyteDB do not allow setting the precision less than the scale explicitly.
```

File: docs/content/preview/migrate/troubleshoot.md [hunk 11]
```diff
@@ -0,0 +1,426 @@
...
+    numeric_max numeric(38,127),
+    float_val FLOAT(5),
+    dec_min_Val dec(1,-84),
+    dec_max_Val dec(38,127),
+    decimal_min_Val decimal(1,-84),
+    decimal_max_Val decimal(38,127)
+);
+```
+
+The exported schema example is as follows:
```
Comment (ddhodge) [inline docs/content/preview/migrate/troubleshoot.md:252]:
```suggestion
The exported schema is as follows:
```

File: docs/content/preview/migrate/troubleshoot.md [hunk 12]
```diff
@@ -0,0 +1,426 @@
...
+
+---
+
+### Issue #584: [Oracle] RAW data fails to get imported in some cases
+
+**GitHub link**: [Issue #584](https://github.com/yugabyte/yb-voyager/issues/584)
+
+**Description**: When attempting to migrate a (LONG) RAW attribute from an Oracle instance, you may face an _invalid hexadecimal error_.
+
+**Workaround**: None. A workaround is being explored currently.
```
Comment (ddhodge) [inline docs/content/preview/migrate/troubleshoot.md:292]:
```suggestion
**Workaround**: None. A workaround is currently being explored.
```

File: docs/content/preview/migrate/troubleshoot.md [hunk 13]
```diff
@@ -0,0 +1,426 @@
...
+    dec_min_val real,
+    dec_max_val decimal,
+    decimal_min_val real,
+    decimal_max_val decimal
+) ;
+```
+
+---
+
+### Issue #584: [Oracle] RAW data fails to get imported in some cases
```
Comment (ddhodge) [inline docs/content/preview/migrate/troubleshoot.md:286]:
```suggestion
### Issue #584: [Oracle] RAW data is not imported in some cases
```

File: docs/content/preview/migrate/troubleshoot.md [hunk 14]
```diff
@@ -0,0 +1,426 @@
...
+
+1. Rename the table name in YugabyteDB to a quoted one using the following command:
+
+    ```sql
+    Alter table test_data_copy2 rename to "test_data_COPY";
+    ```
+
+---
+
+### Issue #207: [Oracle] Some numeric types do not get exported
```
Comment (ddhodge) [inline docs/content/preview/migrate/troubleshoot.md:226]:
```suggestion
### Issue #207: [Oracle] Some numeric types are not exported
```

File: docs/content/preview/migrate/troubleshoot.md [hunk 15]
```diff
@@ -0,0 +1,426 @@
...
+
+Suggested change to the schema is to remove the back-ticks as follows:
+
+```sql
+CREATE INDEX exp_ind ON exp_index_test ((extract(year from date(to_date))));
+```
+
+---
+
+### Issue #320: [MySQL] Issue when exporting data from MySQL with table_name including quotes
```
Comment (ddhodge) [inline docs/content/preview/migrate/troubleshoot.md:164]:
```suggestion
### Issue #320: [MySQL] Exporting data from MySQL when table names include quotes
```

File: docs/content/preview/migrate/troubleshoot.md [hunk 16]
```diff
@@ -0,0 +1,426 @@
...
+
+**Workaround**: None. A workaround is being explored currently.
+
+---
+
+### Issue #602: [Oracle] Issue using a variation of `trunc` function with datetime column in Oracle and YugabyteDB
+
+**GitHub link**: [Issue #602](https://github.com/yugabyte/yb-voyager/issues/602)
+
+**Description**: You can use the `trunc` function with a timestamp column in your Oracle schema, but this variation is not supported in YugabytedB, because the `date_trunc` function is used for such types of datetime columns. So, when you export such a schema using `trunc`, it exports `trunc` and fails during data import.
```
Comment (ddhodge) [inline docs/content/preview/migrate/troubleshoot.md:300]:
```suggestion
**Description**: You can use the `trunc` function with a timestamp column in your Oracle schema, but this variation is not supported in YugabytedB, where the `date_trunc` function is used for these types of datetime columns. When you export such a schema using `trunc`, the data import fails.
```

File: docs/content/preview/migrate/troubleshoot.md [hunk 17]
```diff
@@ -0,0 +1,426 @@
...
+
+**Workaround**: Manual intervention needed. You have to replace the function from `trunc` to `date_trunc` in the exported schema files.
```
Comment (ddhodge) [inline docs/content/preview/migrate/troubleshoot.md:302]:
```suggestion
**Workaround**: Manual intervention needed. You have to replace `trunc` with `date_trunc` in the exported schema files.
```

File: docs/content/preview/migrate/troubleshoot.md [hunk 18]
```diff
@@ -0,0 +1,426 @@
...
+---
+
+## Unsupported features
+
+Currently, yb-voyager doesn't support the following features:
+
+| Feature | Description/Alternatives  | GitHub Issue |
+| :------ | :------------------------ | :----------- |
+| ALTER VIEW | YugabyteDB does not yet support any schemas containing `ALTER VIEW` statements. | [48](https://github.com/yugabyte/yb-voyager/issues/48) |
+| BLOB and CLOB | yb-voyager currently ignores all columns of type BLOB/CLOB. <br>  Use another mechanism to load the attributes till this feature is supported.| [43](https://github.com/yugabyte/yb-voyager/issues/43) |
```
Comment (ddhodge) [inline docs/content/preview/migrate/troubleshoot.md:426]:
```suggestion
| BLOB and CLOB | yb-voyager currently ignores all columns of type BLOB/CLOB. <br>Use another mechanism to load the attributes.| [43](https://github.com/yugabyte/yb-voyager/issues/43) |
```

File: docs/content/preview/migrate/troubleshoot.md [hunk 19]
```diff
@@ -0,0 +1,426 @@
...
+
+Note that the DDL gets exported with `trunc` function and you have to replace it with `date_trunc` after export as follows:
+
+```sql
+ALTER TABLE test_timezone ADD CONSTRAINT test_cc1 CHECK ((dtts = date_trunc('day',dtts)));
+```
+
+---
+
+### Issue #334: [MySQL, Oracle] Issue when importing with case-sensitive schema names
```
Comment (ddhodge) [inline docs/content/preview/migrate/troubleshoot.md:320]:
```suggestion
### Issue #334: [MySQL, Oracle] Importing case-sensitive schema names
```

File: docs/content/preview/migrate/troubleshoot.md [hunk 20]
```diff
@@ -0,0 +1,426 @@
...
+
+**GitHub link**: [Issue #584](https://github.com/yugabyte/yb-voyager/issues/584)
+
+**Description**: When attempting to migrate a (LONG) RAW attribute from an Oracle instance, you may face an _invalid hexadecimal error_.
+
+**Workaround**: None. A workaround is being explored currently.
+
+---
+
+### Issue #602: [Oracle] Issue using a variation of `trunc` function with datetime column in Oracle and YugabyteDB
```
Comment (ddhodge) [inline docs/content/preview/migrate/troubleshoot.md:296]:
```suggestion
### Issue #602: [Oracle] Using a variation of `trunc` with datetime columns in Oracle and YugabyteDB
```

**Review by ddhodge — APPROVED** [review]

File: docs/content/preview/migrate/troubleshoot.md [hunk 21]
```diff
@@ -207,7 +207,7 @@ DBD::mysql::st execute failed: Table 'pk_missing.test_data_COPY' doesn't exist a
 
 Suggested workaround is as follows:
 
-1. Rename the table name with quotes to a name without quotes in MySQL database using the following command:
+1. Rename the table with the quotes using a name without quotes in MySQL database using the following command:
```
Comment (ddhodge) [inline docs/content/preview/migrate/troubleshoot.md:210]:
```suggestion
1. In MySQL, rename the table using a name without quotes using the following command:
```

**Review by shubham-yb — APPROVED** [review]

**Pull request merged**